gdpr/impl_test.go

```diff
@@ -837,3 +838,154 @@ func TestTCF1Consent(t *testing.T) {
 	assert.Equal(t, false, passGeo, "TCF1 consent - passing geo not allowed")
 	assert.Equal(t, false, passID, "TCF1 consent - passing id not allowed")
 }
+
+func TestAllowActivitiesVendorException(t *testing.T) {
+	noPurposeOrVendorConsentAndPubRestrictsP2 := "CPF_61ePF_61eFxAAAENAiCAAAAAAAAAAAAAACEAAAACEAAgAgAA"
+	noPurposeOrVendorConsentAndPubRestrictsNone := "CPF_61ePF_61eFxAAAENAiCAAAAAAAAAAAAAACEAAAAA"
+
+	testDefs := []struct {
+		description           string
+		p2VendorExceptionMap  map[openrtb_ext.BidderName]struct{}
+		sp1VendorExceptionMap map[openrtb_ext.BidderName]struct{}
+		bidder                openrtb_ext.BidderName
+		consent               string
+		allowBid              bool
+		passGeo               bool
+		passID                bool
+	}{
+		{
+			description:          "Bid/ID blocked by publisher - p2 enabled with p2 vendor exception, pub restricts p2 for vendor",
```

Comment:
Does a vendor exception not override the TCF2 consent string value? ... or is that not what's going on in this test?

Reply:
A vendor exception overrides a consent string value but a publisher restriction beats a vendor exception. This is testing that the publisher restriction prevents the bid/id even though the vendor exception is set.